sycl/source/detail/event_impl.cpp

```diff
@@ -174,21 +175,31 @@ event_impl::event_impl(ur_event_handle_t Event, const context &SyclContext)
   }
 }
 
-event_impl::event_impl(const QueueImplPtr &Queue)
-    : MQueue{Queue}, MIsProfilingEnabled{!Queue || Queue->MIsProfilingEnabled} {
-  if (Queue)
-    this->setContextImpl(Queue->getContextImplPtr());
-  else {
-    MState.store(HES_NotComplete);
+event_impl::event_impl(queue_impl &Queue, private_tag)
+    : MQueue{Queue.weak_from_this()},
+      MIsProfilingEnabled{Queue.MIsProfilingEnabled} {
+  this->setContextImpl(Queue.getContextImplPtr());
+  MState.store(HES_Complete);
```

Comment:
That's how it was before my changes, but I have no idea why. If anyone has any idea, I'd like to hear it.